lib/backend/sqlite.cc

```diff
@@ -194,8 +208,19 @@ static int sqlite_fini(rpmdb rdb)
 	    rdb->db_opens--;
 	} else {
 	    if (sqlite3_db_readonly(sdb, NULL) == 0) {
+
 		sqlexec(sdb, "PRAGMA optimize");
 		sqlexec(sdb, "PRAGMA wal_checkpoint = TRUNCATE");
+
+		int max_size = rpmExpandNumeric("%{?_sqlite_vacuum}");
+		if (max_size <= 0)
+		    max_size = 20*1024*1024;
+		long free_space = sqlite_free_space(sdb);
```

Comment:
On 32bit architectures, long is a mere 32bit integer. Better be safe and use explicit int64_t for the size since that's you're getting it from sqlite to begin with.

Reply:
Yeah, I have been pondering about int vs long vs int64_t here.

rpmExpandNumeric also only returns an `int`. I wonder if we should at least move the C++ variant to 64 bits. Not that we need it here urgently but adding a 32 bit API seems kinda wrong.